src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt

```diff
@@ -208,7 +211,40 @@ internal class RecommendationService(
     validateRecallType(jsonRequest)
     val existingRecommendationEntity = getRecommendationEntityById(recommendationId)
 
-    return updateAndSaveRecommendation(jsonRequest, existingRecommendationEntity, userId, readableUserName, null, false, false, pageRefreshIds, featureFlags)
+    return updateAndSaveRecommendation(
+      jsonRequest,
+      existingRecommendationEntity,
+      userId,
+      readableUserName,
+      null,
+      false,
+      false,
+      pageRefreshIds,
+      featureFlags
+    )
+  }
+
+  @OptIn(ExperimentalStdlibApi::class)
+  suspend fun updateRecommendationWithManagerRecallDecision(
+    jsonRequest: JsonNode?,
+    recommendationId: Long,
+    readableUserName: String?
+  ): RecommendationResponse {
+    validateManagerRecallDecision(jsonRequest)
+    val existingRecommendationEntity = getRecommendationEntityById(recommendationId)
+    val userAccessResponse = existingRecommendationEntity.data.crn?.let { userAccessValidator.checkUserAccess(it) }
+    if (userAccessValidator.isUserExcludedRestrictedOrNotFound(userAccessResponse)) {
+      throw UserAccessException(Gson().toJson(userAccessResponse))
+    } else {
+      val readerForUpdating: ObjectReader = CustomMapper.readerForUpdating(existingRecommendationEntity.data)
```

Comment:
This block here is copied from the `updateAndSaveRecommendation` method. Wonder if it's worth creating a method of the common bits to make it abit cleaner and reduce copied code, WDYT?

Reply:
Sounds good. I did look at that, but Intellij didn't offer any quick auto refactor solution.
Are you ok if I take that task on in my next tciket?